ansible-ipi-install/roles/installer/tasks/60_deploy_ocp.yml

```diff
@@ -17,7 +17,7 @@
   block:
     - name: Run OpenShift Cluster install as async task
       shell: |
-        /usr/local/bin/openshift-baremetal-install --dir {{ dir }} --log-level debug create cluster
+        /usr/local/bin/openshift-baremetal-install --dir {{ dir }} --log-level debug create cluster > {{ dir }}/create-cluster.stdout 2> {{ dir }}/create-cluster.stderr
```

Comment:
Thanks for your contribution @mulbc but I thought the logs are captured in /home/kni/.openshift_install.log on the provisioner host, by default. How is this different?

Reply:
That file does not exist for me...

```
[root@e24-h23-740xd ~]# cd /home/kni/
[root@e24-h23-740xd kni]# ls -al
total 28
drwx------  8 kni  kni   220 Jul 20 15:31 .
drwxr-xr-x. 3 root root   17 Jul 20 10:25 ..
drwx------  3 kni  kni    17 Jul 20 10:25 .ansible
drwxrwxr-x  2 kni  kni   254 Jul 20 15:36 .ansible_async
drwxr-xr-x  8 root root  233 Jul 20 12:47 badfish
-rw-------  1 kni  kni   347 Jul 20 11:43 .bash_history
-rw-r--r--  1 kni  kni    18 Aug 30  2019 .bash_logout
-rw-r--r--  1 kni  kni   141 Aug 30  2019 .bash_profile
-rw-r--r--  1 kni  kni   371 Jul 20 10:43 .bashrc
drwx------  4 kni  kni    44 Jul 20 10:43 .cache
drwxr-xr-x  4 kni  kni  4096 Jul 20 16:28 clusterconfigs
-rw-r--r--  1 root root  245 Jul 20 13:10 hosts.txt
drwxr-xr-x  2 kni  kni    80 Jul 20 11:34 .ssh
-rw-r--r--  1 root root   98 Jul 20 13:35 workers.txt
[root@e24-h23-740xd kni]# ls -al /home/kni/.openshift_install.log
ls: cannot access '/home/kni/.openshift_install.log': No such file or directory
```

Reply:
My bad, it's /home/kni/clusterconfigs/.openshift_install.log

Reply:
That's a bit too late - I don't have the scale lab environment any more... but I assume that log is written once Ansible is done, not while Ansible is working?
If so, my change will allow the user to watch the openshift-install output during the install to see at what stage the install is, if the IPI is doing some actions currently or to immediately see where errors are.

For me this was very nice and I thought others would enjoy it as well

Reply:
Sorry about the delay, I was busy in other OpenStack related projects. No, the log is written realtime when ansible is running.

Reply:
Do you get the output of openshift-install in that log?

Example:
```DEBUG OpenShift Installer 4.5.4                    
DEBUG Built from commit 01f5643a02f154246fab0923f8828aa9ae3b76fb 
DEBUG Fetching Install Config...                   
DEBUG Loading Install Config...                    
DEBUG   Loading SSH Key...                         
DEBUG   Loading Base Domain...                     
DEBUG     Loading Platform...                      
DEBUG   Loading Cluster Name...                    
DEBUG     Loading Base Domain...                   
DEBUG     Loading Platform...                      
DEBUG   Loading Pull Secret...                     
DEBUG   Loading Platform...                        
DEBUG Using Install Config loaded from state file  
DEBUG Reusing previously-fetched Install Config    
INFO Waiting up to 30m0s for the cluster at https://api.bench.chris.ocs.ninja:6443 to initialize... 
DEBUG Still waiting for the cluster to initialize: Cluster operator console is reporting a failure: RouteHealthDegraded: route not yet available, https://console-openshift-console.apps.bench.chris.ocs.ninja/health returns '503 Service Unavailable' 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: downloading update 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 2% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 7% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 11% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 12% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 43% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 67% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 72% complete 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 86% complete 
DEBUG Still waiting for the cluster to initialize: Cluster operator marketplace has not yet reported success 
DEBUG Still waiting for the cluster to initialize: Working towards 4.5.4: 87% complete 
```

if so, I'll close this ticket

Reply:
Yes,
```time="2020-08-04T15:42:44Z" level=debug msg="Built from commit 01f5643a02f154246fab0923f8828aa9ae3b76fb"
time="2020-08-04T15:42:44Z" level=debug msg="Fetching Install Config..."
time="2020-08-04T15:42:44Z" level=debug msg="Loading Install Config..."
time="2020-08-04T15:42:44Z" level=debug msg="  Loading SSH Key..."
time="2020-08-04T15:42:44Z" level=debug msg="  Loading Base Domain..."
time="2020-08-04T15:42:44Z" level=debug msg="    Loading Platform..."
time="2020-08-04T15:42:44Z" level=debug msg="  Loading Cluster Name..."
time="2020-08-04T15:42:44Z" level=debug msg="    Loading Base Domain..."
time="2020-08-04T15:42:44Z" level=debug msg="    Loading Platform..."
time="2020-08-04T15:42:44Z" level=debug msg="  Loading Pull Secret..."
time="2020-08-04T15:42:44Z" level=debug msg="  Loading Platform..."
time="2020-08-04T15:42:44Z" level=debug msg="Using Install Config loaded from state file"
time="2020-08-04T15:42:44Z" level=debug msg="Reusing previously-fetched Install Config"
time="2020-08-04T15:42:44Z" level=info msg="Waiting up to 1h0m0s for the cluster at https://api.test167.myocp4.com:6443 to initialize..."
time="2020-08-04T15:42:45Z" level=debug msg="Still waiting for the cluster to initialize: Working towards 4.5.4: 69% complete"
time="2020-08-04T15:42:45Z" level=debug
time="2020-08-04T15:42:45Z" level=debug msg="Terraform has been successfully initialized!"
time="2020-08-04T15:42:45Z" level=debug
time="2020-08-04T15:42:45Z" level=debug msg="You may now begin working with Terraform. Try running \"terraform plan\" to see"
time="2020-08-04T15:42:45Z" level=debug msg="any changes that are required for your infrastructure. All Terraform commands"
time="2020-08-04T15:42:45Z" level=debug msg="should now work."
time="2020-08-04T15:42:45Z" level=debug
time="2020-08-04T15:42:45Z" level=debug msg="If you ever set or change modules or backend configuration for Terraform,"
time="2020-08-04T15:42:45Z" level=debug msg="rerun this command to reinitialize your working directory. If you forget, other"
time="2020-08-04T15:42:45Z" level=debug msg="commands will detect it and remind you to do so if necessary."
/Built from

```

Apologies for the wrong location of the file earlier and delay in getting back. Thanks for taking the time to contribute!